woocommerce/woocommerce
# fix: set correct shipping_method indexes in cart

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.


### Changes proposed in this Pull Request:




The `$index` variable is never set within `templates/cart/cart-shipping.php` is never set and so all of the shipping methods have `name=shipping_method[0]` and `woocommerce_after_shipping_rate` always receives `0` for the index.
<img width="470" alt="Screenshot 2024-06-11 at 14 48 59" src="https://github.com/woocommerce/woocommerce/assets/2420216/1137fdb5-69d2-4e45-ae77-de721b0497bb">

This change sets `$index` correctly within the loop:
<img width="483" alt="Screenshot 2024-06-11 at 14 51 02" src="https://github.com/woocommerce/woocommerce/assets/2420216/f05bbb42-4fd9-4257-ba3d-c7e74b715680">




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `WooCommerce > Settings > Shippings` and configure a Shipping Zone with multiple shipping methods available. 
2. Add at least one product to your basket.
3. Go to the Basket or Checkout page and fill in the billing address details until the shipping methods populate.
4. The shipping method `input` elements should have sequential name attributes of the form `shipping_method[i]` where `i` is their position in the list. You can use your browser's dev tools to verify this.



### Changelog entry

Will update manually.
